lighthouse-extension/app/src/lighthouse-background.js

```diff
@@ -138,6 +140,39 @@ window.runLighthouseInExtension = function(options, categoryIDs) {
     });
 };
 
+/**
+ * @param {string} url
+ * @param {!Object} options Lighthouse options.
+          Specify lightriderFormat to change the output format.
+ * @param {!Array<string>} categoryIDs Name values of categories to include.
+ * @return {!Promise}
+ */
+window.runLighthouseInWebRendering = function(url, options, categoryIDs) {
+  // Default to 'info' logging level.
+  log.setLevel('info');
+  const connection = window.newWebRenderingConnection();
+  const startTime = Date.now();
+  return window.runLighthouseForConnection(connection, url, options, categoryIDs)
+    .then(results => {
+      const endTime = Date.now();
+      results.timing = {total: endTime - startTime};
+      const artifacts = results.artifacts;
+      filterOutArtifacts(results);
+      let promise = Promise.resolve();
+      if (options && options.saveAssets) {
+        promise = assetSaver.logAssets(artifacts, results.audits);
+      }
+      promise.then( _ => {
+        if (options && options.lightriderFormat === 'json') {
+          connection.quit(JSON.stringify(results));
```

Comment:
probably a bit more clear to do a `const output = json ? stringify(results) : generatereport()`  and then  `connection.quit(output)`.

does connection.quit() return a promise we should return?